app/models/schedule.rb

```diff
@@ -0,0 +1,14 @@
+class Schedule < ActiveRecord::Base
+  has_and_belongs_to_many :order_cycles, join_table: 'order_cycle_schedules'
+  has_many :coordinators, uniq: true, through: :order_cycles
+
+  attr_accessible :name, :order_cycle_ids
+
+  validates :order_cycles, presence: true
+
+  scope :with_coordinator, lambda { |enterprise| joins(:order_cycles).where('coordinator_id = ?', enterprise.id).select('DISTINCT schedules.*') }
```

Comment:
Might be a little clearer if we specify `order_cycles.coordinator_id` here? Not that it's an ambiguous column, but it took me a little while to figure out where the column was coming from.